src/java_bytecode/java_utils.cpp

```diff
@@ -21,11 +21,9 @@ Author: Daniel Kroening, kroening@kroening.com
 
 bool java_is_array_type(const typet &type)
 {
-  if(type.id()!=ID_struct)
+  if(type.id() != ID_struct)
     return false;
-  return has_prefix(id2string(
-    to_struct_type(type).get_tag()),
-                    "java::array[");
+  return is_java_array_tag(to_struct_type(type).get_tag());
```

Comment:
I didn't know about this method - I wonder if all instances should be replaced with `java_types::is_valid_java_array` (this is out of scope so not required!)